jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java

```diff
@@ -708,20 +718,22 @@ public Collection<MetaClass> getAllJpaEmbeddableMetaClasses() {
     }
 
     /**
-     * @return collection of all Java enums used as a type of an entity attribute
+     * @return collection of all Java enums implementing {@link io.jmix.core.metamodel.datatype.EnumClass}
      */
     public Collection<Class<?>> getAllEnums() {
         if (enums == null) {
             synchronized (this) {
-                enums = new HashSet<>();
-                for (MetaClass metaClass : metadata.getSession().getClasses()) {
-                    for (MetaProperty metaProperty : metaClass.getProperties()) {
-                        if (metaProperty.getRange().isEnum()) {
-                            Class<?> c = metaProperty.getRange().asEnumeration().getJavaClass();
-                            enums.add(c);
-                        }
+                Set<Class<?>> tmpEnums = new HashSet<>();
+                Set<String> classNames = classpathScanner.getClassNames(EnumDetector.class);
+                for (String className : classNames) {
+                    try {
+                        Class<?> enumClass = ReflectionHelper.loadClass(className);
+                        tmpEnums.add(enumClass);
+                    } catch (ClassNotFoundException e) {
+                        log.warn("Cannot load enum class {}", className, e);
                     }
                 }
+                enums = tmpEnums;
```

Comment:
Maybe we should return unmodifiable set?

```suggestion
                enums = Collections.unmodifiableSet(tmpEnums);
```